src/components/utils/test/data_accessor_test.cc

```diff
@@ -52,17 +52,17 @@ TEST(DataAccessorTest, CreateDataAccessor) {
 TEST(DataAccessorTest, CreateDataAccessor_MutexIsLocked_CannotLockItAgain) {
   // arrange
   int test_value = 10;
-  sync_primitives::Lock testSet_lock_;
+  sync_primitives::Lock testSet_lock_(true);
   DataAccessor<int> testdata(test_value, testSet_lock_);
 
   // assert
-  EXPECT_FALSE(testSet_lock_.Try());
+  EXPECT_TRUE(testSet_lock_.Try());
```

Comment:
@OHerasym You changed the logic of test. Mutex should not be recursive and trying to lock should finish with fail.